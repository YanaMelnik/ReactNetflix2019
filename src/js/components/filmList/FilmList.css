.film-list {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    align-items: flex-start;
    align-content: stretch;
    margin-top: 40px;
    margin-bottom: 40px;
}

.film-element {
    flex: 0 0 28%;
    margin-bottom: 30px;
    font-family: sans-serif;
    max-width: 28%;
}
