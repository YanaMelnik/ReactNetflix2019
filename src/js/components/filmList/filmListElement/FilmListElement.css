.film-element__header {
    display: flex;
    justify-content: space-between;
    text-transform: uppercase;
    margin: 10px 10px 0;
}

.film-element__title {

}

.film-element__year {
    padding: 1px 5px;
    border: 1px solid black;
    font-size: 11px;
    border-radius: 3px;
    max-height: 13px;
    min-height: 13px;
}

.film-element__genre {
    margin: 0 10px;
    color: #8d8d8d;
    font-size: 12px;
}
