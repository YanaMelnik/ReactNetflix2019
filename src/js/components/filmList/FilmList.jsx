import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FilmListElement from './filmListElement/FilmListElement.jsx';
import './FilmList.css';

function filmMapCallback(elem) {
  return <FilmListElement
      key={elem.id}
      filmName={elem.title}
      yearOfIssue={elem.release_date}
      genre={elem.genres[0]}
      filmPosterUrl={elem.poster_path}/>;
}

class FilmList extends Component {
  render() {
    const filmListArray = this.props.filmsArray.map(filmMapCallback);
    return (
      <div className='wrapper film-list'>
        {filmListArray}
      </div>
    );
  }
}

FilmList.propTypes = {
  filmsArray: PropTypes.array.isRequired,
};

export default FilmList;
