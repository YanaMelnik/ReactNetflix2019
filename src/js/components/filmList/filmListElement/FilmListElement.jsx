import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FilmPoster from '../../common/filmPoster/FilmPoster';

import './FilmListElement.css';

class FilmListElement extends Component {
  render() {
    return (
      <div>
        <FilmPoster filmName={this.props.filmName} imgUrl={this.props.filmPosterUrl}/>

        <div className='film-element__header'>
          <span className='film-element__title'>{this.props.filmName}</span>
          <span className='film-element__year'>{this.props.yearOfIssue}</span>
        </div>

        <span className='film-element__genre'>{this.props.genre}</span>
      </div>
    );
  }
}

FilmListElement.propTypes = {
  filmName: PropTypes.string.isRequired,
  yearOfIssue: PropTypes.number.isRequired,
  genre: PropTypes.string.isRequired,
  filmPosterUrl: PropTypes.string.isRequired,
};

export default FilmListElement;
