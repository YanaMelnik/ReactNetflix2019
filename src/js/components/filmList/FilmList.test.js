import React from 'react';
import { shallow } from 'enzyme';
import FilmList from './FilmList';
import FilmListElement from './filmListElement/FilmListElement';
import expect from 'expect';

describe('FilmList', () => {
    it('should render component correctly in case when all props defined', () => {
        const film = {
            id: 1,
            title: 'test',
            release_date: 1991,
            genres: ['drama'],
            poster_path: 'test'
        };
        const component = shallow(<FilmList filmsArray={[film]}/>);

        expect(component).toMatchSnapshot();
    });

    it('should render component correctly in when films are not defined', () => {
        const component = shallow(<FilmList/>);

        expect(component).toMatchSnapshot();
    });

    it('should render film list component in case when films are defined', () => {
        const film = {
            id: 1,
            title: 'test',
            release_date: 1991,
            genres: ['drama'],
            poster_path: 'test'
        };
        const component = shallow(<FilmList filmsArray={[film]}/>);

        expect(component.find(FilmListElement)).toHaveLength(1);
    });

    it('should render special text in case when films are not defined', () => {
        const component = shallow(<FilmList/>);

        expect(component.find('div.no_content')).toHaveLength(1);
    });
});
