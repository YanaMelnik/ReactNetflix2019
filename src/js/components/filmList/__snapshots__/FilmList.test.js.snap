// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList should render component correctly in case when all props defined 1`] = `
<div
  className="wrapper film-list"
>
  <FilmListElement
    filmName="test"
    filmPosterUrl="test"
    genre="drama"
    key="1"
    yearOfIssue={1991}
  />
</div>
`;

exports[`FilmList should render component correctly in when films are not defined 1`] = `
<div
  className="wrapper film-list"
>
  <div
    className="no_content"
  >
    No films found
  </div>
</div>
`;
